internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -271,6 +277,29 @@ func (s *LoadBalancedWebService) Template() (string, error) {
 	return string(overriddenTpl), nil
 }
 
+func (s *LoadBalancedWebService) containerPortMappings() ([]*template.PortMapping, []*template.PortMapping, error) {
+	var primaryPorts []manifest.ExposedPort
+	var sidecarPorts []manifest.ExposedPort
+	exposedPorts, err := s.manifest.ExposedPorts()
+	if err != nil {
+		return nil, nil, err
+	}
+	// Sort the exposed ports so that the order is consistent and the integration test won't be flaky.
+	sort.Slice(exposedPorts, func(i, j int) bool {
+		return exposedPorts[i].Port < exposedPorts[j].Port
+	})
+
+	for _, portConfig := range exposedPorts {
+		if portConfig.ContainerName == s.name {
+			primaryPorts = append(primaryPorts, portConfig)
+		} else {
+			sidecarPorts = append(sidecarPorts, portConfig)
+		}
+	}
+
+	return ConvertPortMapping(primaryPorts), ConvertPortMapping(sidecarPorts), nil
+}
```

Comment:
Can we make it a separate function to reduce duplication?

Reply:
Sorry but I did not understand what duplication you are referring to in here? Could you please elaborate?

Reply:
everything after we get all the exposed ports from the manifest https://github.com/aws/copilot-cli/pull/4192/files#diff-7e36dd6cfea6499215e00fda7bd632bec4a52ffec8428ebe43981553b39c0cd2R287-R300 because they'll not be relavant to the manifest type anymore